java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java

```diff
@@ -37,9 +38,15 @@
 public class RangeEqualsVisitor {
 
   protected final ValueVector right;
-  protected final int leftStart;
-  protected final int rightStart;
-  protected final int length;
+  protected int leftStart;
+  protected int rightStart;
+  protected int length;
+
+  protected boolean needCheckType = true;
+
+  public void setNeedCheckType(boolean needCheckType) {
```

Comment:
can you add a unit test for this set to `false` with non-dictionary vectors?

Reply:
Fixed.